### ISSUE TITLE
Enable headless server hosting; Remove GUI dependencies from TES3MP server

### ISSUE THREAD

**Issue by Lazalatin** [TES3MP/TES3MP#342]
Hi folks,

ever since i started using TES3MP I was wondering if I could be able to host my TES3MP server on my headless Ubuntu server. Unfortunately, this is not possible right now since TES3MP requires Qt5 and a GUI in order to be started and properly configured. This is a No-No IMHO, so I open this issue in hope to raise attention to this matter. I refuse to install a graphical framework on my headless server in order to run a server binary which runs in CLI on my GUI enabled Desktop at home. So I am sure there is a way to get rid of GUI dependencies for the TES3MP server.

Of course I am aware that removing all GUI from the TES3MP server and implementing CLI abilities for it in order to configure and run is not a small task. So I hope I can contribute to this issue myself, if I have some spare time. If there is a way to fulfill my needs right now, hours of online searching did not bring it up.

If anyone else is able and willing to activate CLI superpowers to TES3MP server binary, I would be glad to support persons doing so with time, beer or cheers, whatever preferred.

If this issue is not appropriate for this project please be kind to a curious mind like mine.
I wish you a good day, evening or night.

Regards,
Lazalatin

**Comment by davidcernat** [comment]
You don't need Qt5 or a GUI at all to run a TES3MP server. Have you tried changing your CMake options so you build just the server?

**Comment by Lazalatin** [comment]
At first I only tried the distributes binary packages, these DO need QT5 to run the server.

I tried building the server myself today and after some tweaks to the command line I got the server running on my headless server. But I would like to emphasize that it would be great to have server-only packagings of tes3mp to be shipped to people without the knowledge of how to build their own server binaries. I will try to create an addition to the builder script to be able to package server-only files.

Maybe we should keep this issue open until some "convenient" way of deploying the server only files for users has been established?
